tests/python/test_reset_ndarrays.py

```diff
@@ -0,0 +1,46 @@
+import gc
+import sys
+
+import pytest
+
+import gstaichi as ti
+
+from tests import test_utils
+
+
+@test_utils.test(arch=[ti.cpu])
+def test_ndarray_reset() -> None:
```

Comment:
It may be a good idea to assert the number of ndarray object being registered?

Reply:
Sure. Can you elaborate a little on what kind of error you are imagining might happen without this? You mean, like, inadvertently registering twice? and/or not registering at all?

Reply:
We want to make sure we can release memory properly, to avoid OOM errors, and just because it should be part of our specifications that unused memory should be reclaimed. In practice, this is important if a user creates a scene, then delete and create another one, repeatedly. This is an actual use case. Not just a thought experiment.

Reply:
Addressed in 54356f1fc